packages/cms/src/provider/container.tsx

```diff
@@ -9,11 +10,98 @@ export function ProviderContainer(props: {
   const settings = getSettings()
   const draft = settings.failure ? false : settings.data.draft
 
+  // TODO: error handling
+  const procedures = {
```

Comment:
At some point, we may want to use https://next-safe-action.dev/ for these sorts of procedures. For now, I'm trying to keep things simple.